currency/pairs.go

```diff
@@ -355,3 +410,33 @@ func (p Pairs) GetStablesMatch(code Code) Pairs {
 	}
 	return stablePairs
 }
+
+// ValidateAndConform checks for duplications and empty pairs then conforms the
+// entire pairs list to the supplied formatting (unless bypassed).
+// Map[string]bool type is used to make sure delimiters are not included so
+// different formatting entry duplications can be found e.g. `LINKUSDTM21`,
+// `LIN-KUSDTM21` or `LINK-USDTM21 are all the same instances but with different
+// unintentional processes for formatting.
+func (p Pairs) ValidateAndConform(pFmt PairFormat, bypassFormatting bool) (Pairs, error) {
```

Comment:
When `bypassFormatting` and when there are duplicate _enabled_ pairs, the following occurs:
```
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[DEBUG] | SYNC | 31/08/2022 12:13:41 | Exchange CurrencyPairSyncer initial sync started. 4 items to process.
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[INFO]  | SYNC | 31/08/2022 12:13:41 | FTX REST BTC-USD SPOT TICKER: Last $20295.00000000 Ask $20295.00000000 Bid $20294.00000000 High $0.00000000 Low $0.00000000 Volume 0.00000000
[DEBUG] | SYNC | 31/08/2022 12:13:41 | FTX ticker sync complete BTC-USD [1/4].
[INFO]  | SYNC | 31/08/2022 12:13:41 | FTX REST SUSHIBEAR-USD SPOT TICKER: Last $0.00000001 Ask $0.00000002 Bid $0.00000001 High $0.00000000 Low $0.00000000 Volume 0.00000000
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[DEBUG] | SYNC | 31/08/2022 12:13:41 | FTX ticker sync complete SUSHIBEAR-USD [2/4].
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[INFO]  | SYNC | 31/08/2022 12:13:41 | FTX REST BTC-USD SPOT ORDERBOOK: Bids len: 100 Amount: 315.593400 BTC. Total value: $6389897.22350000 Asks len: 100 Amount: 297.835600 BTC. Total value: $6061319.67010000
[DEBUG] | SYNC | 31/08/2022 12:13:41 | FTX orderbook sync complete BTC-USD [3/4].
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[DEBUG] | PORTFOLIO | 31/08/2022 12:13:41 | Portfolio: Updating FTX USD entry with balance 3832.962144.
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | LOG | 31/08/2022 12:13:41 | FTX failed to get cryptocurrency deposit address for SUSHIBEAR. Err: deposit address does not exist SUSHIBEAR 
[INFO]  | SYNC | 31/08/2022 12:13:41 | FTX REST SUSHIBEAR-USD SPOT ORDERBOOK: Bids len: 1 Amount: 1030000000.000000 SUSHIBEAR. Total value: $10.30000000 Asks len: 100 Amount: 21541000000.000000 SUSHIBEAR. Total value: $86486.09000000
[DEBUG] | SYNC | 31/08/2022 12:13:41 | FTX orderbook sync complete SUSHIBEAR-USD [4/4].
[DEBUG] | SYNC | 31/08/2022 12:13:41 | Exchange CurrencyPairSyncer initial sync is complete.
[DEBUG] | SYNC | 31/08/2022 12:13:41 | Exchange CurrencyPairSyncer initial sync took 394.512736ms [4 sync items].
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:41 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[INFO]  | LOG | 31/08/2022 12:13:42 | Deposit addresses synced
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
[ERROR] | SYNC | 31/08/2022 12:13:42 | FTX failed to get enabled pairs. Err: 1INCH-PERP currency pair duplication of asset type futures
```

Reply:
That's not ideal.